src/pages/index.js

```diff
@@ -1,67 +1,263 @@
+  .getUserAvatar()
+  .then((data) => {
+    userAvatar.setUserAvatar({
+      avatar: data.avatar,
+    });
+  })
+  .catch((error) => {
+    console.error("Failed to fetch profile Avatar:", error);
+  });
+
+// -------------------------------------------------------------------------------------
+// Editing UserAvatar Popup
+// -------------------------------------------------------------------------------------
+const avatarEditPopup = new PopupWithForm("#avatar-edit-modal", {
+  handleFormSubmit: (formData) => {
+    const avatarURL = formData["Image URL"];
+    avatarEditPopup.toggleUploadIndicator(true);
+
+    api
+      .editUserAvatar({ avatar: avatarURL })
+      .then((data) => {
+        userAvatar.setUserAvatar({
+          avatar: data.avatar,
+        });
+        avatarEditPopup.close();
+      })
+      .catch((error) => {
+        console.log("Error updating Avatar:", error);
+      })
+      .finally(() => {
+        avatarEditPopup.toggleUploadIndicator(false);
+      });
+  },
+});
+
+avatarEditPopup.setEventListeners();
+
+// -------------------------------------------------------------------------------------
+// Card Section Instance
+// -------------------------------------------------------------------------------------
+const cardSection = new Section(
+  {
+    items: [],
+    renderer: (cardData) => {
+      const card = new Card(
+        cardData,
+        cardTemplate,
+        (cardData) => imagePopup.open(cardData),
+        handleTrashClick,
+        updateLike
+      ).getView();
+      if (cardData.isLiked) {
+        card
+          .querySelector(".card__like-button")
+          .classList.add("card__like-button_active");
+      }
+      cardSection.appendItem(card);
+    },
+  },
+  ".cards__list"
+);
+
+// -------------------------------------------------------------------------------------
+// Initial Card Creation & Rendering from API
+// -------------------------------------------------------------------------------------
+api
+  .getCardList()
+  .then((cardList) => {
+    cardSection.renderItems(cardList);
+  })
+  .catch((error) => {
+    console.error("Failed to fetch or render cards:", error);
+  });
+
+// -------------------------------------------------------------------------------------
+// Additional Card adding to API & DOM
+// -------------------------------------------------------------------------------------
 const addCardPopup = new PopupWithForm("#add-card-modal", {
   handleFormSubmit: (formData) => {
     const cardData = {
       name: formData["title"],
       link: formData["Image URL"],
     };
-    const card = createCard(cardData);
-    cardSection.prependItem(card);
-    addCardPopup.getForm().reset();
-    formValidators[
-      addCardPopup.getForm().getAttribute("name")
-    ].toggleButtonState();
-    addCardPopup.close();
+
+    addCardPopup.toggleUploadIndicator(true);
+
+    api
+      .addCard(cardData)
+      .then((apiCardData) => {
+        const card = new Card(
```

Comment:
You should make a special function `createCard` for creating a card not to copy-paste the code when you need to add a new card with `append`.

```js

function createCard(item) {
    // here will be the code of creating
  return cardElement.getView();
}

```